### PR TITLE
PEP 637 allow for unpacking of starargs

### PR DESCRIPTION
**Pull request by stefanoborini**
As from @eric-wieser idea on python-ideas, *args are now unpacked.

### CODE REVIEW

**Comment by stefanoborini** [comment]
PR built on top of #1620 . Need merge of that one first.

File: pep-0637.rst [hunk 1]
```diff
@@ -347,17 +347,12 @@ The successful implementation of this PEP will result in the following behavior:
    - but if no ``**kwargs`` parameter is defined, it is an error.
 
 
-7. Sequence unpacking remains a syntax error inside subscripts::
+7. Sequence unpacking is allowed inside subscripts::
```
Comment (eric-wieser) [inline pep-0637.rst:350]: Note that this requires some specification for what `obj[*()]` should do, which likely has similar semantics to `obj[]`. One option would be an error, another would be to handle it like `obj[kwargs=...]` is handled.

Comment (eric-wieser) [reply pep-0637.rst:350]: Also would probably want to include `obj[*(x,)] == obj[x]` and `obj[*(x,),] == obj[x,]` as corner cases.

**Comment by gvanrossum** [comment]
@stefanoborini Are you going to respond to @eric-wieser's latest comments or should I just merge like this?

**Comment by stefanoborini** [comment]
@gvanrossum no he has a good point. I'll add those cases (and solve conflicts)

File: pep-0637.rst [hunk 2]
```diff
@@ -354,12 +354,34 @@ The successful implementation of this PEP will result in the following behavior:
    This allows notations such as ``[:, *args, :]``, which could be treated 
    as ``[(slice(None), *args, slice(None))]``.
 
+   The following notation equivalence should be honored::
+
+       obj[*()]
```
Comment (gvanrossum) [inline pep-0637.rst:359]: Did you discuss this on python-ideas? I actually think the reason for rejecting `obj[]` (that it looks like a typo) doesn't apply here. Ditto for `obj[**{}]`.

Comment (eric-wieser) [reply pep-0637.rst:359]: I'm in favor of allowing that too - my point above was mainly that _something_ needed to be specified, and then python-ideas could argue for a better specification.

**Review by gvanrossum — COMMENTED** [review]
Okay, LGTM. I'll land when tests pass.

**Comment by gvanrossum** [comment]
That test is just hanging. I'm going to land.

File: pep-0637.rst (hunk 2 above)
Comment (stefanoborini) [reply pep-0637.rst:359]: I was not sure about it... maybe we should let it pass, and treat it like obj[()], same for obj[**{}] is valid and equivalent to obj[()]

**Pull request merged**